__init__.py

```diff
@@ -220,28 +200,27 @@ def on_volume_set(self, message):
         vol = message.data.get("percent", 0.5)
         vol = clip(vol, 0.0, 1.0)
 
-        self.volume = vol
         self.muted = False
-        self.set_hardware_volume(vol)
+        self.volume.set_hardware_volume(vol)
         self.show_volume = True
 
     def on_volume_get(self, message):
         """ Handle request for current volume. """
-        self.bus.emit(message.response(data={'percent': self.volume,
+        self.bus.emit(message.response(data={'percent': self.volume.get_hardware_volume(),
                                              'muted': self.muted}))
         self.show_volume = message.data.get('show', False)
 
     def on_volume_duck(self, message):
         """ Handle ducking event by setting the output to 0. """
         self.muted = True
         self.mute_pulseaudio()
-        self.set_hardware_volume(0)
+        self.volume.set_hardware_volume(0)
 
     def on_volume_unduck(self, message):
         """ Handle ducking event by setting the output to previous value. """
         self.muted = False
         self.unmute_pulseaudio()
-        self.set_hardware_volume(self.volume)
+        self.volume.set_hardware_volume(self.volume.volume)
```

Comment:
So technically I wanted the volume class to handle both system level volume and hardware volume. In this case it would be beneficial to have a system_volume vs hardware_volume attribute but I was taking a minimalistic approach here. I'll rename it hardware_volume for now.